impress-org/givewp-next-gen
# Refactor: disable update button until dirty

## Description



Form Builder updates:
- Change "Publish" button to "Update"
- Add simple `isDirty` state and disable update button until `isDirty` is true.
- Clean up `BlockEditorContainer` and move InterfaceSkeleton into `BlockEditorInterfaceSkeletonContainer`
- Fix a few react warnings.
- Update currency component with types

Note:
- The `isDirty` state is a very basic implementation and is only set after updating the form.  Later we might decide to do an actual dynamic comparison of the previous / changed state to determine its dirty-ness.

## Affects



The form builder

## Visuals



![2023-02-14 15 09 46](https://user-images.githubusercontent.com/10138447/218850806-35f512f6-a181-45f1-9f15-647cbad3b2f1.gif)


## Testing Instructions



- pull branch and run `npm run build` (*note: I noticed running `npm run dev/watch:form-builder` instead of build would sometimes make the block provider not recognize the _first_ change in some cases which affects updating & dirty state.)
- change blocks/form settings and use the update button
- the update button should be disabled after using the update button.

## Pre-review Checklist
